### ISSUE TITLE
Gauge uses hardcoded specs folder (at least in the logs)

### ISSUE THREAD

**Issue by hakito** [getgauge/gauge#302]
Though specifying a custom path in tha gauge command for the specs folder

`gauge another_specs_folder`

The logs indicate it still tries to load specs from a `specs` folder.

api.log:

`
02:06:41.492 Unable to add directory C:\work\specs to file watcher: GetFileAttributes: The system cannot find the file specified.
`

**Comment by kaustavdm** [comment]
Changes implemented. As a side effect, there have been 2 changes to gauge's usage:

1. `--daemonize` now takes optional arguments for each spec directory that it needs to watch. It defaults to `spec` as the spec directory if nothing is provided. e.g.

      ```sh
      $ GAUGE_API_PORT=12345 gauge --daemonize spec_dir_1/ spec_dir_2/ spec_dir_3/
      ```
 
2. `--refactor` now takes optional arguments for spec directories after specifying step names. e.g.:

      ```sh
      $ gauge --refactor "Old step text" "New step text" spec_dir_1/ spec_dir_2/ spec_dir_3/
      ```

**Comment by mahendrakariya** [comment]
We should have functional tests for this. 
Created an issue. https://github.com/getgauge/gauge-tests/issues/11

**Comment by mahendrakariya** [comment]
More fixes for this issue in abc90586af10d8b1b.
